gage_outages/all_gages_map.R

```diff
@@ -0,0 +1,153 @@
+  ref <- elide(ref, scale=scale, center=orig.cent, bb = bbox(ref))
+  new.cent <- rgeos::gCentroid(ref, byid=TRUE)@coords
+  obj <- elide(obj, shift=shift*10000+c(orig.cent-new.cent))
+  if (is.null(proj.string)){
+    proj4string(obj) <- proj4string(sp)
+  } else {
+    proj4string(obj) <- proj.string
+  }
+  
+  if (!is.null(row.names)){
+    row.names(obj) <- row.names
+  }
+  return(obj)
+}
+
+# thanks to Bob Rudis (hrbrmstr):
+# https://github.com/hrbrmstr/rd3albers
+
+# -- if moving any more states, do it here: --
+move_variables <- list(
+  AK = list(scale=0.33, shift = c(80,-450), rotate=-50),
+  HI = list(scale=1, shift=c(520, -110), rotate=-35)
+)
+
+stuff_to_move <- list(
+  AK = to_sp("world", "USA:alaska"),
+  HI = to_sp("world", "USA:hawaii")
+)
+
+conus <- to_sp('state')
+states.out <- conus
+wgs84 <- "+init=epsg:4326"
+coords = cbind(all_sites_locations$dec_long_va, all_sites_locations$dec_lat_va)
+
+sites = SpatialPoints(coords, proj4string = CRS(wgs84)) %>%
+  spTransform(CRS(proj4string(states.out)))
+
+sites.df <- as.data.frame(sites)
+
+for(i in names(move_variables)){
+  shifted <- do.call(shift_sp, c(sp = stuff_to_move[[i]],
+                                 move_variables[[i]],
+                                 proj.string = proj4string(conus),
+                                 row.names = i))
+  states.out <- rbind(shifted, states.out, makeUniqueIDs = TRUE)
+  
+  shifted.sites <- do.call(shift_sp, c(sp = sites[all_sites_locations$state == i,],
+                                       move_variables[[i]],
+                                       proj.string = proj4string(conus),
+                                       ref=stuff_to_move[[i]])) %>%
+    as.data.frame %>%
+    coordinates()
+  
+  sites.df[all_sites_locations$state == i, ] <- shifted.sites
+  
+}
+
+sites.df_plot <- bind_cols(sites.df, all_sites_locations) %>% 
+  mutate(Affected = ifelse(is.na(Affected), yes= "Not affected", no = Affected))
+########
+
+gsMap <- ggplot() +
+  geom_polygon(aes(x = long, y = lat, group = group),
+               data = states.out, fill = "grey90",
+               alpha = 0.9, color = "grey") +
+  coord_sf(datum=NA) +
+  geom_polygon(aes(x = long, y = lat, group = group),
+               data = states.out, fill = NA,
+               alpha = 0.9, color = "grey") +
+  geom_point(data = sites.df_plot, size = 0.5, aes(x = coords.x1, y=coords.x2, color = Affected)) +
+  theme(panel.grid = element_blank(),
+        axis.text = element_blank(),
+        axis.title = element_blank(),
+        plot.title = element_text(hjust = 0.5),
+        plot.subtitle = element_text(hjust = 0.5),
+        panel.background = element_rect(fill = "transparent", color = NA),
+        legend.key = element_rect(fill = "transparent", color = NA),
+        legend.text=element_text(size=12)) + 
+  guides(color=guide_legend(title="")) +
+  ggtitle("Site Outage Summary")
+ggsave(plot = gsMap, filename = "all_gages_outage.png")
```

Comment:
These seemed to plot in portrait mode (in the email) - do you need to specify width and height to make them landscape?

Reply:
yeah I've fixed that